src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -67,11 +70,48 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
             override fun visitArrayType(o: RsArrayType) = collectDiagnostics(holder, o)
             override fun visitVariantDiscriminant(o: RsVariantDiscriminant) = collectDiagnostics(holder, o)
             override fun visitPolybound(o: RsPolybound) = checkPolybound(holder, o)
+            override fun visitTryExpr(o: RsTryExpr) = checkTryExpr(holder, o)
         }
 
         element.accept(visitor)
     }
 
+    private fun checkTryExpr(holder: AnnotationHolder, o: RsTryExpr) {
+        val items = StdKnownItems.relativeTo(o)
```

Comment:
Somewhat inconsistent. Consider something like this:
```
val items = StdKnownItems.relativeTo(o)
val lookup = ImplLookup.relativeTo(o)
val tryTrait = items.findTryTrait() ?: return
val fromTrait = items.findFromTrait() ?: return
val tryExprType = o.expr.type

val errorType = findErrorTyUsingTryTrait(tryExprType, tryTrait, lookup)
if (errorType == null) { ... }

val returnType = findParentFnOrLambdaRetTy(o)
val fooErrorTy = returnType?.let { findErrorTyUsingTryTrait(it, tryTrait, lookup) }
```

Reply:
Done